backend/vectorize_utils.ml

```diff
@@ -95,3 +95,37 @@ module Vectorized_instruction = struct
       results = Array.init res_count (fun i -> Result i)
     }
 end
+
+let vectorizable_machtypes (r1 : Reg.t) (r2 : Reg.t) =
+  match r1.typ, r2.typ with
+  | (Vec128 | Valx2), (Val | Int | Addr | Float | Float32 | Vec128 | Valx2)
+  | (Val | Int | Addr | Float | Float32), (Vec128 | Valx2) ->
+    Misc.fatal_errorf "Unexpected vector machtype Vec128 or Valx2: %a %a"
+      Printreg.reg r1 Printreg.reg r2
+  | Val, Val -> true
+  | Val, (Int | Addr | Float | Float32) | (Int | Addr | Float | Float32), Val ->
+    false
+  | (Int | Addr | Float | Float32), (Int | Addr | Float | Float32) ->
+    (* It is safe to mix Float32, Float, and Int for the purpose of GC, because
+       they are not scannable. It may not be possible to vectorize the
+       operation. *)
```

Comment:
`Addr` here worries me a bit. An `Addr` value is something like an internal pointer into the middle of a heap block - not only are they not GC'able, the GC must not be allowed to run at all while they are live (as they can be invalidated by the GC moving things). So combining, say, an `Int` and an `Addr` into a `Vec128` sounds bad, as you'd lose the fact that the GC must be inhibited while this value is live.

Reply:
For now we can just not vectorize instructions that use registers of type `Addr`. Would it be enough to address this concern? (A better solution would be to introduce `Addrx2` but I would like to do it as a follow after we find an example where this is useful and probably refactor `machtype_component` to make it easier to have a vector of components.)

Reply:
Yes, I would just not vectorize those.

Reply:
Done